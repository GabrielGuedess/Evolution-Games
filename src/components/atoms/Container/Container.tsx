import styled, { css } from 'styled-components';

import { customMedia } from 'utils/media/customMedia';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.grid.container};
    margin: 0 1.6rem;

    ${customMedia.greaterThan('large')`
      margin: 0 3.2rem;
    `}

    ${customMedia.greaterThan('huge')`
      margin: 0 4.8rem;
    `}

    ${customMedia.greaterThan('max')`
      margin-left: auto;
      margin-right: auto;
    `}
  `}
`;
