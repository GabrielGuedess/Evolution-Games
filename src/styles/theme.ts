export default {
  grid: {
    container: '165.6rem',
  },
  border: {
    radius: {
      regular: '1rem',
    },
  },
  font: {
    family: {
      primary:
        "'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    },
    regular: 400,
    medium: 500,
    sizes: {
      xxsmall: '1rem',
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '1.8rem',
      xxmedium: '2.0rem',
      big: '2.2rem',
      xbig: '2.5rem',
      xxbig: '2.8rem',
      xxxbig: '3.0rem',
      large: '3.2rem',
      xlarge: '4.5rem',
      xxlarge: '5.0rem',
      extra: '6.0rem',
    },
  },
  colors: {
    primary: '#7C3AED',
    secondary: '#717B8A',
    mainBg: '#04040C',
    white: '#FFFFFF',
    whiteText: '#F4F4F5',
    gameDetails: '#010914',
    gameGreat: '#0FBF00',
    gameAverage: '#FFF720',
    gameBad: '#FF3333',
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    big: '5.8rem',
    huge: '8.0rem',
    section: '10rem',
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
  shadows: {
    text: '0 0.8rem 1rem rgba(118, 74, 241, 0.5)',
    box: '0 0.8rem 2.8rem rgba(118, 74, 241, 0.5)',
  },
} as const;
