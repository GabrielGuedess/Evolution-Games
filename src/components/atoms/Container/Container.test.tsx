import theme from 'styles/theme';

import { renderWithTheme } from 'utils/tests/helpers';

import { Container } from './Container';

describe('<Container />', () => {
  it('Deve renderizar o componente <Container />', () => {
    // Arrange
    const { container } = renderWithTheme(<Container />);

    // Assert
    expect(container.firstChild).toHaveStyleRule(
      'max-width',
      theme.grid.container,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
