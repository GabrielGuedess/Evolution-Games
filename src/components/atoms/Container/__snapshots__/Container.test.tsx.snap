// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Container /> Deve renderizar o componente <Container /> 1'] = `
.c0 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

@media (min-width:1170px) {
  .c0 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c0 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c0 {
    margin-left: auto;
    margin-right: auto;
  }
}

<div
  class="c0"
/>
`;
